dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataset/DataSetService.java

```diff
@@ -304,4 +305,15 @@ LockStatus getLockStatus( User user, DataElement dataElement, Period period, Org
      * @return a list of LockException with given filter list
      */
     List<LockException> filterLockExceptions( List<String> filters );
+
+    /**
+     * Return a mapping between the given data sets and the associated
+     * organisation units. Only data sets for which the current user has data
+     * write sharing access to are returned.
+     *
+     * @param dataSetUids the data set identifiers.
+     * @return a {@link SetValuedMap} between data sets and organisation unit
+     *         identifiers.
+     */
+    SetValuedMap<String, String> getDataSetOrganisationUnitsAssociations();
```

Comment:
Do we want to keep using the special collections? I'd thought we could use `Map<String, Set<String>>` since the building now can be written nicer using `computeIfAbsent`?

Reply:
Agreed. I just didn't want to change the underlying JdbcOrgUnitAssociationsStore interface in order to avoid a growing task. We could change this later.